sound/soc/intel/boards/sof_sdw.c

```diff
@@ -1394,13 +1394,13 @@ static int create_sdw_dailink(struct snd_soc_card *card, int *link_index,
 		if (cpu_dai_id[i] != ffs(adr_link_next->mask) - 1)
 			continue;
 
-		/* sanity check */
-		if (*codec_conf_index + adr_link_next->num_adr - adr_index > codec_count) {
-			dev_err(dev, "codec_conf: out-of-bounds access requested\n");
-			return -EINVAL;
-		}
-
 		for (j = adr_index; j < adr_link_next->num_adr; j++) {
+			/* sanity check */
+			if (*codec_conf_index >= codec_count) {
+				dev_err(dev, "codec_conf array overflowed\n");
+				return -EINVAL;
+			}
```

Comment:
Agree with @ujfalusi it's best to keep the code as is. I don't see what this simplifies.

Reply:
I will look into it but the out of loop check feels pretty hard to get right, certainly the current one looks iffy. I stopped short of actually working out if it was broken or not, but it does index + num_adr - adr_index, which felt like it was assuming all the devices after the current one on the current soundwire link would be included in the DAI which is likely to not always be true. The in loop check is simple and always checks we have space for the one CODEC we are adding on that loop.

Reply:
Ok, so once the endpoints check is moved into the inner loop, this check also needs to be in the inner loop. It is basically ok outside the loop at the moment, because a lot of other things blow up if you have multiple endpoints groups in a single link at the moment. I will update the commit message to make this all more clear.